extensions/amp-analytics/0.1/analytics-root.js

```diff
@@ -288,19 +313,61 @@ export class AnalyticsRoot {
    * @param {?string=} selectionMethod Allowed values are `null`,
    *   `'closest'` and `'scope'`.
    * @param {boolean=} opt_multiSelectorOn multi-selector expriment
-   * @return {!Promise<!AmpElement>} AMP element corresponding to the selector if found.
+   * @return {!Promise<!Array<!AmpElement>>} Array of AMP elements corresponding to the selector if found.
    */
-  getAmpElement(context, selector, selectionMethod, opt_multiSelectorOn) {
-    return this.getElement(context, selector, selectionMethod).then(element => {
+  getAmpElementOrElements(
+    context,
+    selector,
+    selectionMethod,
+    opt_multiSelectorOn
+  ) {
+    let elementsPromise = this.getElement(context, selector, selectionMethod);
+
+    // Return unique elements based upon selector
+    if (opt_multiSelectorOn) {
       userAssert(
-        element.classList.contains('i-amphtml-element'),
-        'Element "%s" is required to be an AMP element',
+        !selectionMethod,
+        'Cannot have selectionMethod defined with an array selector: %s',
         selector
       );
-      return element;
+      elementsPromise = this.getElementsByScopedQuerySelectorAll_(
+        selector
+      ).then(elements => {
+        const uniqueElements = [];
+        for (let i = 0; i < elements.length; i++) {
+          for (let j = 0; j < elements[i].length; j++) {
```

Comment:
`getElementByScopedQuerySelectorAll_` returns an array of elements, why when would it be a nested array?

Reply:
followed up offline about how to check for unique elements